### PR TITLE
Yes

### PR DESCRIPTION
**Pull request by upatel99**
The yes command is still a work in progress. Currently, I am working on what test cases to include. Here is somewhat of a rough draft of the code and test cases.

### CODE REVIEW

File: cmds/core/yes/yes.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+)
+
+func main() {
+	args := os.Args
```
Comment (pallaud) [inline cmds/core/yes/yes.go]: I think like Ron said earlier, using flag.Args() would be the best option here because it would take care of a lot of checks for you and all you'd have to do is deal with the actual arguments being passed in
So you should be able to call `flag.Parse()` and then use `flag.Args()` to get the arguments - the slice returned should only consist of actual arguments to the yes cmd, see https://gobyexample.com/command-line-flags for an example

File: cmds/core/yes/yes.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	yOutput := "y"
+	if len(args) == 0 {
```
Comment (pallaud) [inline cmds/core/yes/yes.go]: Since you have a default string to print when the user doesn't give any arguments ("y"), it wouldn't be an error if `len(flag.Args())` was 0; instead maybe what you could do is check if `len(flag.Args()) > 0` and if it is, replace `yOutput` with whatever the user provided, and if not, continue with the default value of `yOutput`

File: cmds/core/yes/yes_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"bufio"
+	//	"bytes"
```
Comment (pallaud) [inline cmds/core/yes/yes_test.go]: Remember to get rid of commented out code at the end!

File: cmds/core/yes/yes.go (hunk 2 above)
Comment (upatel99) [reply cmds/core/yes/yes.go]: Sounds good, I just fixed it. I realized using flag.Args definitely resolved many of my checks.

File: cmds/core/yes/yes.go (hunk 1 above)
Comment (upatel99) [reply cmds/core/yes/yes.go]: Yup, I just fixed that :). Thank you!

File: cmds/core/yes/yes_test.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		log.Fatal(err)
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 20; i++ {
+		if in.Scan() {
+			log.Println(in.Text())
+		} else {
+			log.Println("No output")
```
Comment (pallaud) [inline cmds/core/yes/yes_test.go]: Maybe `t.Fatalf` here because if you're not getting output from the yes cmd, you want to log and fail the test right then

File: cmds/core/yes/yes_test.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 20; i++ {
+		if in.Scan() {
+			log.Println(in.Text())
```
Comment (pallaud) [inline cmds/core/yes/yes_test.go]: Instead of printing `in.Text()` here, maybe check that the output is equal to what you expect, in this case "y"; similarly, I think you'll want to add another test that calls the yes command with some arguments, and then you'd want to make sure that `in.Text()` matches the arguments you put in
Looking at testutil.Command, you should just be able to add the arguments by doing `testutil.Command(t, "arg1", "arg2")`

File: cmds/core/yes/yes_test.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+		log.Fatal(err)
+	}
+
+	if err := cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 20; i++ {
```
Comment (pallaud) [inline cmds/core/yes/yes_test.go]: Probably increase this to be 100 iterations or something

File: cmds/core/yes/yes_test.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
+	cmd := testutil.Command(t)
+	fmt.Println(cmd.Path)
```
Comment (pallaud) [inline cmds/core/yes/yes_test.go]: You can delete this print if it was for debugging

File: cmds/core/yes/yes_test.go (hunk 6 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: yup, I was thinking about increasing the iterations to a 1000 :)!

File: cmds/core/yes/yes.go [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+import (
+	"flag"
+	"fmt"
+	"strings"
+)
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+	yOutput := "y"
```
Comment (rminnich) [inline cmds/core/yes/yes.go]: I think calling it yes is fine, not yOutput. Even y would be fine. Go tends to emphasize terse names :-)

File: cmds/core/yes/yes.go [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+	for {
+		if len(args) > 0 {
```
Comment (rminnich) [inline cmds/core/yes/yes.go]: This should be outside the for loop

File: cmds/core/yes/yes_test.go (hunk 7 above)
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: don't use fmt.Print of any kind in a test, use t.Logf

File: cmds/core/yes/yes_test.go (hunk 5 above)
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: don't use log. anything, use t.Logf

File: cmds/core/yes/yes_test.go (hunk 4 above)
Comment (rminnich) [reply cmds/core/yes/yes_test.go]: don't; use the got/want pattern, e.g.
t.Errorf("Reading output: got no input, want input") for example from Go source:
t.Errorf("got %d want %d", got, want)

Since you are iterating here, go ahead and run it for 10 or 20 times, but then you can use t.Errorf for each failure.

File: cmds/core/yes/yes_test.go [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 20; i++ {
+		if in.Scan() {
+			log.Println(in.Text())
+		} else {
+			log.Println("No output")
+		}
+	}
+	cmd.Process.Kill()
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: What if it fails? You need to check the error return.

Comment (rminnich) [inline cmds/core/yes/yes_test.go]: if err := cmd.Process.Kill(); err != nil {
t.Errorf("Killing child process: got %v, want nil", err)
}

File: cmds/core/yes/yes_test.go (hunk 5 above)
Comment (rminnich) [reply cmds/core/yes/yes_test.go]: t.Errorf("scan: got failure, want no error")
break

Comment (rminnich) [reply cmds/core/yes/yes_test.go]: i := in.Text()
if i != "y" {
log.Errorf("in.Text: got %v, want y", i)
break
}

File: cmds/core/yes/yes.go (hunk 8 above)
Comment (upatel99) [reply cmds/core/yes/yes.go]: Very convenient :)!

File: cmds/core/yes/yes_test.go (hunk 4 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it!

File: cmds/core/yes/yes_test.go (hunk 5 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed this! I added another function, which tested the yes command if I passed in multiple parameters. In this case, I passed in "hi how are you", and tested it across 1000 iterations to see if each iteration, I received the output "hi how are you".

File: cmds/core/yes/yes.go (hunk 9 above)
Comment (upatel99) [reply cmds/core/yes/yes.go]: Fixed this!

File: cmds/core/yes/yes_test.go (hunk 7 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: I realized the fmt.Println(cmd.Path) was not necessary to keep, so I just deleted that line of code :)!

File: cmds/core/yes/yes_test.go (hunk 5 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed!

File: cmds/core/yes/yes_test.go (hunk 10 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Oh, this makes sense. This is a very good point. Thank you!

File: cmds/core/yes/yes_test.go [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
+	cmd := testutil.Command(t)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		t.Logf("Could not return a pipe that will be connected to the command's standard output, received err: %v", err)
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: If you can't get a pipe. you can't continue. You should use t.Fatalf here.

File: cmds/core/yes/yes_test.go (hunk 10 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Correct, I checked if the process is actually being killed or not.

File: cmds/core/yes/yes_test.go [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	if err := cmd.Start(); err != nil {
+		t.Log(err)
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: Again, I'm not sure you can continue after this. use t.Fatal

File: cmds/core/yes/yes_test.go [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 1000; i++ {
+		if in.Scan() && in.Text() != "y" {
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: If in.Scan returns false, you just loop 1000 times without checking anything, where you should instead be failing if it fails to scan.

Something like this:
if !in.Scan() {
    t.Fatalf(//some error message)
}
if text := in.Text(); text != "y" {
    t.Errorf(//some error message)
}

File: cmds/core/yes/yes_test.go [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+			t.Errorf("Expected a 'y', but received %v", in.Text())
+			break
+		}
+	}
+	if err := cmd.Process.Kill(); err != nil {
+		t.Errorf("Killing the child process: received %v, but wanted nil", err)
+	}
+}
+
+func TestYesArgs(t *testing.T) {
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: As above. I would actually suggest you compress these two tests into one test that can call yes with various arguments. Right now almost all the test logic is duplicated beyond the args and the string. One example to look at is pkg/ebda/ebda_test.go. Look at how we can test the same functionality with different arguments or setup. You could do something similar

Comment (rminnich) [reply cmds/core/yes/yes_test.go]: gan, I agree, but can we get it working and then merge?

**Comment by rminnich** [comment]
urvisha do a git pull --rebase to sync up your branch

File: cmds/core/yes/yes_test.go (hunk 12 above)
Comment (rminnich) [reply cmds/core/yes/yes_test.go]: t.Fatalf("cmd.Start: got %v, want nil", err)

File: cmds/core/yes/yes_test.go (hunk 11 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Yup, that is a good point. Fatal is stop the execution at that point. Thank you for pointing it out to me :)!

File: cmds/core/yes/yes_test.go (hunk 12 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it

File: cmds/core/yes/yes_test.go (hunk 13 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: That is also a very good point. I need to test that condition first just to make sure I am not looping through 1000 times even if it does not scan anything! Thank you, Gan.

File: cmds/core/yes/yes_test.go (hunk 14 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: I just pushed my recent updates. The code works. If there are any other updates needed to be made, I will go ahead and merge my two test functions together :)! Thank you.

File: cmds/core/yes/yes.go [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+
+func main() {
+	flag.Parse()
+	args := flag.Args()
+	yes := "y"
+	if len(args) > 0 {
+		yes = strings.Join(args, " ")
+	}
+	for {
+		fmt.Println(yes)
```
[the remark this answers is not in the document]
Comment (upatel99) [reply cmds/core/yes/yes.go]: That is a good catch. I made sure to do the error checking just in case if println does give an error.

File: cmds/core/yes/yes.go [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+func main() {
+	flag.Parse()
+	args := flag.Args()
+	yes := "y"
+	if len(args) > 0 {
+		yes = strings.Join(args, " ")
+	}
+	for {
+		if _, err := fmt.Println(yes); err != nil {
+			log.Fatalf("Println got %v error, want nil", err)
```
Comment (rminnich) [inline cmds/core/yes/yes.go]: thinking about this, I was wrong about adding a log.Fatal. You can try this experiment
yes | echo hi
echo does not read from stdin so yes write to stdout fails first time. There is no error printed from yes.
yes should silently exit.

you can see the error with this command
strace -e trace=write yes | echo hi
write(1, "y\ny\ny\ny\ny\ny\ny\ny\ny\ny\ny\ny\ny\ny\ny\ny\n"..., 8192) = -1 EPIPE (Broken pipe)

yes does not print an error, just exits, so I gave you bad guidance, just change that log.Fatalf line to break.
sorry!

File: cmds/core/yes/yes_test.go [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("cmd.Start: got %v, want nil", err)
+	}
+
+	in := bufio.NewScanner(stdout)
+
+	for i := 0; i < 1000; i++ {
+		if !in.Scan() {
+			t.Fatalf("Could not scan: got %v error, want no error", in.Err())
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: just got %v not got %v error
just want nil, not want no error

File: cmds/core/yes/yes_test.go [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+		}
+		if text := in.Text(); text != "y" {
+			t.Errorf("Expected a 'y', but received %v", text)
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: "Expected %v, want 'y'", text

File: cmds/core/yes/yes_test.go [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+			break
+		}
+	}
+	if err := cmd.Process.Kill(); err != nil {
+		t.Errorf("Killing the child process: received %v, but wanted nil", err)
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: got %v, want nil
The reason to fix on this format is then it's easy to look for all these types of prints:
git grep got.*want

File: cmds/core/yes/yes_test.go (hunk 18 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed

File: cmds/core/yes/yes_test.go (hunk 17 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed. I will get used to this format from now on :)!

File: cmds/core/yes/yes.go (hunk 16 above)
Comment (upatel99) [reply cmds/core/yes/yes.go]: I see what you are saying. I fixed it.

File: cmds/core/yes/yes_test.go (hunk 19 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed :)! I will get used to this format!

File: cmds/core/yes/yes_test.go [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
+	cmd := testutil.Command(t)
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		t.Fatalf("cmd.StdoutPipe: got %v, want nil", err)
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: Not really an issue, but you could stand to have something more descriptive than just cmd.StdoutPipe. Something like "Failed to get stdout of yes command: got %v ..." instead.

File: cmds/core/yes/yes_test.go [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+	}
+
+	if err := cmd.Start(); err != nil {
+		t.Fatalf("cmd.Start: got %v, want nil", err)
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: Likewise

File: cmds/core/yes/yes_test.go [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+	}
+	in := bufio.NewScanner(stdout)
+	for i := 0; i < 1000; i++ {
+		if !in.Scan() {
+			t.Fatalf("Could not scan: got %v, want nil", in.Err())
+		}
+		if text := in.Text(); text != "y" {
+			t.Errorf("Got %v, want 'y'", text)
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: nit: might be useful to print the iteration where it failed.

File: cmds/core/yes/yes_test.go (hunk 20 above)
Comment (rminnich) [reply cmds/core/yes/yes_test.go]: that's my fault, sorry. Sure make it more verbose.

File: cmds/core/yes/yes_test.go (hunk 22 above)
Comment (rminnich) [reply cmds/core/yes/yes_test.go]: good idea.

File: cmds/core/yes/yes_test.go (hunk 20 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it :)!

File: cmds/core/yes/yes_test.go (hunk 21 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it :)!

File: cmds/core/yes/yes_test.go (hunk 22 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: That is a very good idea

File: cmds/core/yes/yes_test.go [hunk 23]
```diff
@@ -0,0 +1,52 @@
...
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+func TestMain(m *testing.M) {
+	testutil.Run(m, main)
+}
+
+func TestYes(t *testing.T) {
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: Looks much better! One more improvement you could make. Take a look at:
https://github.com/linuxboot/fiano/blob/57266e68c811cb6cf84f1dc89c85a5bdbb1525f4/pkg/uefi/section_test.go#L26

Notice how we can give names to tests and run them? this allows you to be more explicit about which subtest failed

File: cmds/core/yes/yes_test.go [hunk 24]
```diff
@@ -0,0 +1,52 @@
...
+		in := bufio.NewScanner(stdout)
+		for i := 0; i < 1000; i++ {
+			if !in.Scan() {
+				t.Fatalf("Could not scan: got %v, want nil", in.Err())
+			}
+			if text := in.Text(); text != v.expected {
+				t.Errorf("Got %v at iteration %d, want %v", text, i, v.expected)
+				break
+			}
+		}
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: This is just about there. 

There's only one problem: it is unusual for yes to get killed; more common for it to exit when the pipe it is writing to closes. We're not testing that latter scenario. One way to test it:
run each test twice, first time, close the stdout pipe and make sure it exits, second time, kill it and make sure it exits.

Another way is to add an element to the struct called
closeorkill bool

and depending on the value of close or kill, either close stdout or kill the child.

File: cmds/core/yes/yes_test.go (hunk 23 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: That is a neat idea. I will definitely start to use this more often when I have table driven tests! :)

File: cmds/core/yes/yes_test.go (hunk 24 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Perfect. As discussed, there are three implementations. One is to run the test twice. Two is to add an element to the struct called closeOrKill bool. Three is to write an entire separate test function and changing the cmd.Process.Kill() to closing stdout pipe and then cmd.Close(). The first implementation would cause me to have three for loops. The third implementation causes me to repeat a lot of the same code, so I decided to go with the second implementation instead. :)

File: cmds/core/yes/yes_test.go (hunk 23 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: This is a great way to get a better idea of where the test fails. I actually failed my tests and it catches each test that failed and tells me the name of each of the failed test. That is very neat. Will start utilizing this more often! Thanks, Gan!

File: cmds/core/yes/yes_test.go [hunk 25]
```diff
@@ -0,0 +1,70 @@
...
+				if text := in.Text(); text != v.expected {
+					t.Errorf("Got %v at iteration %d, want %v", text, i, v.expected)
+					break
+				}
+			}
+			if v.closeOrKill == true {
+				if err := stdout.Close(); err != nil {
+					t.Fatalf("Close standard out pipe: got %v, but want nil", err)
+				}
+				v.closeOrKill = false
```
Comment (GanShun) [inline cmds/core/yes/yes_test.go]: why is this line necessary? remove

File: cmds/core/yes/yes_test.go [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+			for i := 0; i < 1000; i++ {
+				if !in.Scan() {
+					t.Fatalf("Could not scan: got %v, want nil", in.Err())
+				}
+				if text := in.Text(); text != v.expected {
+					t.Errorf("Got %v at iteration %d, want %v", text, i, v.expected)
+					break
+				}
+			}
+			if v.closeOrKill == true {
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: so one last thing, the name closeOrKill confused me. Is it true for close, or true for kill. I would recommend 
closeStdout

File: cmds/core/yes/yes_test.go (hunk 25 above)
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: I don't see why you need to set it to false.

File: cmds/core/yes/yes_test.go [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+				if err := stdout.Close(); err != nil {
+					t.Fatalf("Close standard out pipe: got %v, but want nil", err)
+				}
+				v.closeOrKill = false
+				if err := cmd.Wait(); err != nil {
+					if err.Error() != "signal: broken pipe" {
+						t.Errorf("Complete the child process: got %v, but want nil", err)
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: got %v, want nil

File: cmds/core/yes/yes_test.go [hunk 28]
```diff
@@ -0,0 +1,70 @@
...
+					t.Fatalf("Could not scan: got %v, want nil", in.Err())
+				}
+				if text := in.Text(); text != v.expected {
+					t.Errorf("Got %v at iteration %d, want %v", text, i, v.expected)
+					break
+				}
+			}
+			if v.closeOrKill == true {
+				if err := stdout.Close(); err != nil {
+					t.Fatalf("Close standard out pipe: got %v, but want nil", err)
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: got %v, want nil

File: cmds/core/yes/yes_test.go [hunk 29]
```diff
@@ -0,0 +1,70 @@
...
+				}
+				v.closeOrKill = false
+				if err := cmd.Wait(); err != nil {
+					if err.Error() != "signal: broken pipe" {
+						t.Errorf("Complete the child process: got %v, but want nil", err)
+					}
+				}
+			} else {
+				if err := cmd.Process.Kill(); err != nil {
+					t.Errorf("Kill the child process: got %v, but want nil", err)
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: got %v, want nil

**Comment by GanShun** [comment]
I also notice you've squashed your commits. Update your commit message, right now it only mentions the tests.

File: cmds/core/yes/yes_test.go [hunk 30]
```diff
@@ -0,0 +1,71 @@
...
+			if v.closeStdout == true {
+				if err := stdout.Close(); err != nil {
+					t.Fatalf("Close standard out pipe: got %v, want nil", err)
+				}
+				if err := cmd.Wait(); err != nil {
+					if err.Error() != "signal: broken pipe" {
+						t.Errorf("Complete the child process: got %v, want 'signal: broken pipe'", err)
+					}
+				} else {
+					t.Error("Complete the child process: got nil, want 'signal:broken pipe'")
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: this is more complex than you need. do this:
if err := cmd.Wait(); err.Error() != "signal: broken pipe" {
t.Errorf("Complete the child process: got %v, want 'signal: broken pipe'", err)
}
you don't need this else part

Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it! :)

File: cmds/core/yes/yes_test.go [hunk 31]
```diff
@@ -0,0 +1,69 @@
...
+					break
+				}
+			}
+			if v.closeStdout == true {
+				if err := stdout.Close(); err != nil {
+					t.Fatalf("Close standard out pipe: got %v, want nil", err)
+				}
+				if err := cmd.Wait(); err != nil {
+					if err.Error() != "signal: broken pipe" {
+						t.Errorf("Complete the child process: got %v, want nil", err)
```
[the remark this answers is not in the document]
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Oops, my bad! I must've missed that!

File: cmds/core/yes/yes_test.go (hunk 29 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it!

File: cmds/core/yes/yes_test.go (hunk 28 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Fixed it! :)

File: cmds/core/yes/yes_test.go (hunk 25 above)
Comment (upatel99) [reply cmds/core/yes/yes_test.go]: Ooops, I see what you're saying! It is unnecessary. Good catch

File: cmds/core/yes/yes_test.go [hunk 32]
```diff
@@ -0,0 +1,67 @@
...
+				if text := in.Text(); text != v.expected {
+					t.Errorf("Got %v at iteration %d, want %v", text, i, v.expected)
+					break
+				}
+			}
+			if v.closeStdout == true {
+				if err := stdout.Close(); err != nil {
+					t.Fatalf("Close standard out pipe: got %v, want nil", err)
+				}
+				if err := cmd.Wait(); err.Error() != "signal: broken pipe" {
```
Comment (rminnich) [inline cmds/core/yes/yes_test.go]: stupid me I made an error
if err := cmd.Wait(); err == nil || err.Error() != "signal: broken pipe" {

if err is nil err.Error will panic.
so this tests if err is nil (that's an error)
or if err.Error != "signal: broken pipe"
which is an error

Further, it will do the tests in the order you write them; it will not reorder them, which is important. 

Sorry

ron
